<?xml version="1.0"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot name="simulator" xmlns:xacro="http://ros.org/wiki/xacro">

  <link name="world"/>
  
  <joint name="virtual_joint" type="fixed">
    <child link="base_link"/>
    <parent link="world"/>
  </joint>

  <link
    name="base_link">
    <inertial>
      <origin
        xyz="-0.00110306076075581 -1.75841246845748E-05 -3.96711644723413E-06"
        rpy="0 0 0" />
      <mass
        value="4.68848447603709" />
      <inertia
        ixx="0.430475145644522"
        ixy="-1.79316182511057E-05"
        ixz="-1.23322016653127E-06"
        iyy="0.430587050486874"
        iyz="4.52382428687083E-07"
        izz="0.592651276900184" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="prop_1_link">
    <inertial>
      <origin
        xyz="2.47996068125644E-13 3.41268162852742E-08 2.2877255645426E-12"
        rpy="0 0 0" />
      <mass
        value="0.0279209337587258" />
      <inertia
        ixx="0.000201162215959647"
        ixy="-1.5059378448931E-17"
        ixz="-3.5479401191167E-05"
        iyy="0.000209948517866518"
        iyz="-9.30869790117904E-17"
        izz="9.10135660869097E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_1_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.294117647058824 0.294117647058824 0.294117647058824 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_1_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="prop_1_joint"
    type="continuous">
    <origin
      xyz="0.181350000000243 -0.314107413952491 0"
      rpy="1.5707963267949 0 -2.80424202552252" />
    <parent
      link="base_link" />
    <child
      link="prop_1_link" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="prop_2_link">
    <inertial>
      <origin
        xyz="-4.91190421669785E-13 -8.46650185456313E-13 -1.36557432028894E-12"
        rpy="0 0 0" />
      <mass
        value="0.0279204568460125" />
      <inertia
        ixx="0.000202041619412603"
        ixy="-1.6765922883602E-17"
        ixz="-3.37223454388012E-05"
        iyy="0.000209943475946723"
        iyz="-7.27420925847124E-17"
        izz="8.21690270950501E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_2_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.294117647058824 0.294117647058824 0.294117647058824 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_2_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="prop_2_joint"
    type="continuous">
    <origin
      xyz="0.362700000000011 0 0"
      rpy="1.5707963267949 0 -1.73577621414184" />
    <parent
      link="base_link" />
    <child
      link="prop_2_link" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="prop_3_link">
    <inertial>
      <origin
        xyz="-2.52770027131533E-13 -3.26235502648694E-08 -3.15525383598469E-13"
        rpy="0 0 0" />
      <mass
        value="0.0279209345793859" />
      <inertia
        ixx="0.000201383717068649"
        ixy="1.22165744202057E-17"
        ixz="-3.5054722379838E-05"
        iyy="0.000209948517974453"
        iyz="1.02203769018779E-17"
        izz="8.87985560643182E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_3_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.294117647058824 0.294117647058824 0.294117647058824 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_3_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="Prop_3_joint"
    type="continuous">
    <origin
      xyz="0.181349999999764 0.314107413952759 0"
      rpy="1.5707963267949 0 2.42964177354604" />
    <parent
      link="base_link" />
    <child
      link="prop_3_link" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="prop_4_link">
    <inertial>
      <origin
        xyz="1.983690989249E-13 3.26215846655616E-08 1.82182047225865E-12"
        rpy="0 0 0" />
      <mass
        value="0.0279209345803512" />
      <inertia
        ixx="0.000200179018418646"
        ixy="6.52149189229152E-17"
        ixz="-3.72576312509752E-05"
        iyy="0.000209948517984384"
        iyz="5.22243367121589E-17"
        izz="1.00845542663739E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_4_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.294117647058824 0.294117647058824 0.294117647058824 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_4_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="prop_4_joint"
    type="continuous">
    <origin
      xyz="-0.181350000000244 0.314107413952479 0"
      rpy="1.5707963267949 0 -2.78817405638384" />
    <parent
      link="base_link" />
    <child
      link="prop_4_link" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="prop_5_link">
    <inertial>
      <origin
        xyz="3.66928709638614E-14 -1.88129293038066E-08 -5.06594766136459E-13"
        rpy="0 0 0" />
      <mass
        value="0.0279204664551623" />
      <inertia
        ixx="0.00020003244164034"
        ixy="3.92299042138082E-18"
        ixz="-3.75010706246379E-05"
        iyy="0.000209943793870452"
        iyz="7.48241521753871E-18"
        izz="1.02263984417856E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_5_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.294117647058824 0.294117647058824 0.294117647058824 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_5_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="prop_5_joint"
    type="continuous">
    <origin
      xyz="-0.362699999999998 0 0"
      rpy="1.5707963267949 0 1.38857142747957" />
    <parent
      link="base_link" />
    <child
      link="prop_5_link" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="prop_6_link">
    <inertial>
      <origin
        xyz="-1.40915057400548E-13 -1.45340183492525E-08 -2.6326163471424E-12"
        rpy="0 0 0" />
      <mass
        value="0.0279209237247354" />
      <inertia
        ixx="0.000198410642206663"
        ixy="-1.25767832104697E-17"
        ixz="-4.00962993381281E-05"
        iyy="0.00020994856295244"
        iyz="6.36819861126385E-17"
        izz="1.18529751758187E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_6_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.294117647058824 0.294117647058824 0.294117647058824 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/prop_6_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="prop_6_joint"
    type="continuous">
    <origin
      xyz="-0.181349999999753 -0.314107413952772 0"
      rpy="1.5707963267949 0 2.43358089892909" />
    <parent
      link="base_link" />
    <child
      link="prop_6_link" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="arm_1_link">
    <inertial>
      <origin
        xyz="-0.000277779744851368 -0.0258931805795906 -0.000458833266730171"
        rpy="0 0 0" />
      <mass
        value="0.0466133022968278" />
      <inertia
        ixx="1.20039983846615E-05"
        ixy="1.03121576615339E-08"
        ixz="-8.26981177401855E-07"
        iyy="7.48803299231645E-06"
        iyz="1.70344296308232E-08"
        izz="1.1138650620346E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/arm_1_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/arm_1_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="arm_1_joint"
    type="revolute">
    <origin
      xyz="0.00640042621662216 -0.0101611186614582 -0.05270000000001"
      rpy="1.5707963267949 0 0" />
    <parent
      link="base_link" />
    <child
      link="arm_1_link" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="arm_2_link">
    <inertial>
      <origin
        xyz="0.0872965795343047 0.0258178427950186 -0.0443985785254201"
        rpy="0 0 0" />
      <mass
        value="0.0881360535329319" />
      <inertia
        ixx="2.03889559871351E-05"
        ixy="1.08647089358152E-08"
        ixz="1.61399453784491E-07"
        iyy="2.8038688798677E-05"
        iyz="1.02875642093698E-08"
        izz="2.41553284315105E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/arm_2_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/arm_2_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="arm_2_joint"
    type="revolute">
    <origin
      xyz="0.013206 -0.0388 0.021814"
      rpy="-1.5708 0.54438 0" />
    <parent
      link="arm_1_link" />
    <child
      link="arm_2_link" />
    <axis
      xyz="0 -1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="arm_3_link">
    <inertial>
      <origin
        xyz="-0.0884957691467715 0.0251944702774581 0.00109148122244698"
        rpy="0 0 0" />
      <mass
        value="0.0841267673801512" />
      <inertia
        ixx="1.54134634493973E-05"
        ixy="1.51426288997905E-08"
        ixz="-6.99845650487514E-07"
        iyy="2.30716087505895E-05"
        iyz="-3.87887886135734E-09"
        izz="2.37159974327882E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/arm_3_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/arm_3_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="arm_3_joint"
    type="revolute">
    <origin
      xyz="0.11089 0.051 -0.068288"
      rpy="-3.1416 0 0" />
    <parent
      link="arm_2_link" />
    <child
      link="arm_3_link" />
    <axis
      xyz="0 -1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="arm_4_link">
    <inertial>
      <origin
        xyz="-0.007441 0.025511 0.045081"
        rpy="0 0 0" />
      <mass
        value="0.093795" />
      <inertia
        ixx="2.2174E-05"
        ixy="-5.361E-08"
        ixz="-2.6295E-06"
        iyy="2.6687E-05"
        iyz="-1.5961E-08"
        izz="3.1188E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/arm_4_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/arm_4_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="arm_4_joint"
    type="revolute">
    <origin
      xyz="-0.124 0 0.0010588"
      rpy="0 0 0" />
    <parent
      link="arm_3_link" />
    <child
      link="arm_4_link" />
    <axis
      xyz="0 -1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="gripper_left_link">
    <inertial>
      <origin
        xyz="-0.0085159 -0.038248 0.025138"
        rpy="0 0 0" />
      <mass
        value="0.01653" />
      <inertia
        ixx="6.782E-06"
        ixy="1.3549E-07"
        ixz="9.1033E-07"
        iyy="9.5271E-06"
        iyz="-1.3399E-07"
        izz="4.3674E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/gripper_left_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.60392 0.64706 0.68627 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/gripper_left_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="gripper_left_joint"
    type="prismatic">
    <origin
      xyz="-0.022963 0.0255 0.06824"
      rpy="0.00085876 0.00052289 -0.0025515" />
    <parent
      link="arm_4_link" />
    <child
      link="gripper_left_link" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="gripper_right_link">
    <inertial>
      <origin
        xyz="-0.0084777 0.041125 0.025151"
        rpy="0 0 0" />
      <mass
        value="0.01653" />
      <inertia
        ixx="6.7753E-06"
        ixy="-5.4166E-08"
        ixz="9.1471E-07"
        iyy="9.5302E-06"
        iyz="1.6154E-07"
        izz="4.3712E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/gripper_right_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.60392 0.64706 0.68627 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulator/meshes/gripper_right_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="gripper_right_joint"
    type="prismatic">
    <origin
      xyz="-0.022963 0.0255 0.06824"
      rpy="0.00085876 0.00052289 -0.0025515" />
    <parent
      link="arm_4_link" />
    <child
      link="gripper_right_link" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>

  <joint name="camera_joint" type="fixed">
    <axis xyz="0 0 0" />
    <origin xyz="0 0 0.5" rpy="0 0 0"/>
    <parent link="base_link"/>
    <child link="camera_link"/>
  </joint>

  <!-- Camera -->
  <link name="camera_link">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <material name="red"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <gazebo reference="camera_link">
    <sensor type="depth" name="camera">
      <update_rate>30.0</update_rate>
      <camera>
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>640</width>
          <height>480</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>camera</cameraName>
        <imageTopicName>/camera/rgb/image_raw</imageTopicName>
        <cameraInfoTopicName>/camera/rgb/camera_info</cameraInfoTopicName>
        <depthImageTopicName>/camera/depth/image_raw</depthImageTopicName>
        <depthImageCameraInfoTopicName>/camera/depth/camera_info</depthImageCameraInfoTopicName>
        <frameName>camera_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
        <CxPrime>0</CxPrime>
        <Cx>318.643040</Cx>
        <Cy>255.313989</Cy>
        <focalLength>516.469215</focalLength>
      </plugin>
    </sensor>
  </gazebo>
</robot>